src/net.h

```diff
@@ -394,17 +394,17 @@ class CConnman
     std::vector<ListenSocket> vhListenSocket;
     std::atomic<bool> fNetworkActive;
     banmap_t setBanned;
-    CCriticalSection cs_setBanned;
+    RecursiveMutex cs_setBanned;
     bool setBannedIsDirty;
     bool fAddressesInitialized;
     CAddrMan addrman;
     std::deque<std::string> vOneShots;
-    CCriticalSection cs_vOneShots;
+    Mutex cs_vOneShots;
```

Comment:
In Bitcoin, RecursiveMutex is used.

Reply:
This is completely removed actually. Bitcoin has reimplemented the peer management layer.